File: BackEnd-TiendaMusical/src/main/java/com/tienda_musical/backendtiendamusical/model/Instrumento.java
package com.tienda_musical.backendtiendamusical.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.util.Set;

@Entity
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class Instrumento extends Master{

    private String instrumento;
    private String marca;
    private String modelo;
    private String imagen;
    private Double precio;
    private Double costoEnvio;
    private Integer cantidadVendida;
    private String descripcion;

    @ManyToOne
    @JoinColumn(name = "categoria_id")
    private Categoria categoria;

    @OneToOne(mappedBy = "instrumento", cascade = { CascadeType.MERGE, CascadeType.PERSIST, CascadeType.REFRESH}, orphanRemoval = true)
    private Stock stock;

    @OneToMany(mappedBy = "instrumento",  cascade = { CascadeType.MERGE, CascadeType.PERSIST, CascadeType.REFRESH})
    private Set<PedidoDetalle> pedidos;

}

File: FrontEnd-TiendaMusical/src/models/Instrumento.ts
import type Categoria from "./Categoria";

export default class Instrumento {
    id?: number;
    instrumento: string = "";
    marca: string = "";
    modelo: string = "";
    imagen: string = "";
    precio: number = 0;
    costoEnvio: number = 0;
    cantidadVendida: number = 0;
    descripcion: string = "";
    categoria?: Categoria;
    eliminado!: boolean;
}
